files/grid-studies-2-1.py

```diff
@@ -7,36 +7,47 @@ class GridStudies(monome.GridApp):
     def __init__(self):
         super().__init__()
 
+    width = 0
+    height = 0
+    step = [[0 for col in range(16)] for row in range(16)]            
```

Comment:
these attributes are declared as class attributes, see this discussion for the differences between class and instance variables:

https://stackoverflow.com/questions/207000/what-is-the-difference-between-class-and-instance-attributes

as we don't need to share them between instances of the app, i'd recommend initializing them on the instance level as `self.width = 0...` in the constructor here and below.

Reply:
ah, makes total sense, thank you! i had tried `self.` ahead of those, but didn't realize the importance of the indentations!